lib/Dialect/FIRRTL/Transforms/Dedup.cpp

```diff
@@ -250,6 +250,12 @@ struct Equivalence {
     if (aType.isa<BundleType>() && bType.isa<BundleType>())
       return check(diag, message, a, aType.cast<BundleType>(), b,
                    bType.cast<BundleType>());
+    if (aType.isa<RefType>() && bType.isa<RefType>())
+      return (check(diag,
+                    message + ", mismatch in data taps with two modules marked "
+                              "with must dedup.",
```

Comment:
Nit: the difference is narrowly in reference types and not "data taps". For this type of thing, you may consider making the message longer with a suggestion of what to fix. E.g.: `mismatch in reference types (did you use a Grand Central Tap or View in a different order?)`. Ideally, we can map the source of the creator of the data tap back to the cause here. Jack's work towards adding source locators to annotations may help bridge this gap until all this moves into Chisel proper.

Reply:
Fixed the error message. Still not incorporated the annotation location info.